go/libraries/doltcore/sqle/commit_ancesctors_table.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"context"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+)
+
+var _ sql.Table = (*CommitAncestorsTable)(nil)
+
+// CommitAncestorsTable is a sql.Table implementation that implements a
+// system table which shows (commit, parent_commit) relationships.
+type CommitAncestorsTable struct {
+	dbName string
+	ddb    *doltdb.DoltDB
+}
+
+// NewCommitAncestorsTable creates a CommitAncestorsTable
+func NewCommitAncestorsTable(ctx *sql.Context, dbName string) (sql.Table, error) {
+	ddb, ok := DSessFromSess(ctx.Session).GetDoltDB(dbName)
+	if !ok {
+		return nil, sql.ErrDatabaseNotFound.New(dbName)
+	}
+
+	return &CommitAncestorsTable{dbName: dbName, ddb: ddb}, nil
+}
+
+// Name is a sql.Table interface function which returns the name of the table.
+func (dt *CommitAncestorsTable) Name() string {
+	return doltdb.CommitAncestorsTableName
+}
+
+// String is a sql.Table interface function which returns the name of the table.
+func (dt *CommitAncestorsTable) String() string {
+	return doltdb.CommitAncestorsTableName
+}
+
+// Schema is a sql.Table interface function that gets the sql.Schema of the commit_ancestors system table.
+func (dt *CommitAncestorsTable) Schema() sql.Schema {
+	return []*sql.Column{
+		{Name: "commit_hash", Type: sql.Text, Source: doltdb.CommitAncestorsTableName, PrimaryKey: true},
+		{Name: "parent_hash", Type: sql.Text, Source: doltdb.CommitAncestorsTableName, PrimaryKey: true},
```

Comment:
This isn't enough -- you also need to be able to order the parent hashes to tell who the merge root was.